src/lazy/text/matched.rs

```diff
@@ -128,7 +150,11 @@ impl MatchedInt {
             }
         };
 
-        Ok(int)
+        if self.is_negative {
+            Ok(-int)
+        } else {
+            Ok(int)
+        }
```

Comment:
🗺️ I had tests to match negative ints, but didn't have tests to read negative ints.